modular_citadel/code/game/machinery/wishgranter.dm

```diff
@@ -0,0 +1,111 @@
+/obj/machinery/wish_granter/attack_hand(mob/living/carbon/user)
+	if(charges <= 0)
+		to_chat(user, "The Wish Granter lies silent.")
+		return
+
+	else if(!ishuman(user))
+		to_chat(user, "You feel a dark stirring inside of the Wish Granter, something you want nothing of. Your instincts are better than any man's.")
+		return
+
+	else if (!insisting)
+		to_chat(user, "Your first touch makes the Wish Granter stir, listening to you.  Are you really sure you want to do this?")
+		insisting++
+
+	else
+		if(is_special_character(user))
+			to_chat(user, "You speak.  [pick("I want power","Humanity is corrupt, mankind must be destroyed", "I want to rule the world","I want immortality")].  The Wish Granter answers.")
+			to_chat(user, "Your head pounds for a moment, before your vision clears. The Wish Granter, sensing the darkness in your heart, has given you limitless power, and it's all yours!")
+			user.dna.add_mutation(HULK)
+			user.dna.add_mutation(XRAY)
+			user.dna.add_mutation(COLDRES)
+			user.dna.add_mutation(TK)
+			user.next_move_modifier *= 0.5	//half the delay between attacks!
+			to_chat(user, "Things around you feel slower!")
+			charges--
+			insisting = FALSE
+			to_chat(user, "You have a very great feeling about this!")
+		else
+			to_chat(user, "The Wish Granter awaits your wish.")
+			var/wish = input("You want...","Wish") as null|anything in list("Power","Wealth","The Station To Disappear","To Kill","Nothing")
+			switch(wish)
```

Comment:
Need to add checking for charges here or it's possible to use one multiple times even when it should have no charges left. Alternatively, add QDELETED(src) checks.

Reply:
Would it not be caught in line 2 of the code, or is this level paranoia required?